### PR TITLE
AO-20841 Introduce CRI plugin as an option for monitoring CRI compliant engines

### PR DESCRIPTION
**Pull request by poleszcz**
I went with the simpliest and the most reliable option - leaving configuration of socket mounting to the user. I've added most expected options in the kustomize yaml.

Proposal of the documentation change:
https://swicloud.atlassian.net/wiki/spaces/~229243988/pages/2828665273/AO-20596+Doc+update+-+proposal

### CODE REVIEW

File: deploy/base/daemonset/kustomization.yaml [hunk 1]
```diff
@@ -13,7 +13,6 @@ configMapGenerator:
     behavior: create
     literals:
       - SWISNAP_ENABLE_APACHE=false
-      - SWISNAP_ENABLE_DOCKER=true
```
Comment (lootek) [inline deploy/base/daemonset/kustomization.yaml:16]: `- SWISNAP_ENABLE_DOCKER=false`

up to you though

File: deploy/base/deployment/kustomization.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ configMapGenerator:
     behavior: create
     literals:
       - SWISNAP_ENABLE_APACHE=false
+      - SWISNAP_ENABLE_CRI=false
```
Comment (lootek) [inline deploy/base/deployment/kustomization.yaml:16]: should this be disabled?

File: deploy/base/daemonset/kustomization.yaml (hunk 1 above)
Comment (poleszcz) [reply deploy/base/daemonset/kustomization.yaml:16]: The default behavior is to use CRI instead od DOCKER. Docker is being removed from base - and proper setting is in the overlay.

File: deploy/base/deployment/kustomization.yaml (hunk 2 above)
Comment (poleszcz) [reply deploy/base/deployment/kustomization.yaml:16]: Yes, I think so. In the "deployment" the primary use case is to monitor whole cluster with kubernetes plugin. 
CRI and Docker are by default disabled. Deamonset is kind of deployment meant to monitor containers.

**Review by lootek — APPROVED** [review]

**Pull request merged**